### PR TITLE
[Do not review] Support OAS3 specs in ActiveDocs

### PR DESCRIPTION
**Pull request by josemigallas**
>⚠️
> This PR is being split into smaller PRs:
> 1. Circleci Fix: #1596 
> 2. Admin Portal: #1594
> 3. Dev Portal: #1595 
> 4. Proper styling: #1597

**What this PR does / why we need it**:

We want to support OAS3

**Which issue(s) this PR fixes** 

[THREESCALE-2725: Start using SwaggerUI 3.18.3 or higher for OAS3 specs](https://issues.redhat.com/browse/THREESCALE-2725)

**Current state**
- **Admin portal**: integration works fine. It's been implemented with `swagger-ui-react`.
Screen Shot 2020-01-17 at 11 57 38

Screen Shot 2020-01-16 at 12 40 52

Screen Shot 2020-01-17 at 11 57 31

- **Dev portal**: integration works fine but styles are a mess. Currently working on a css patch to make it look like v2
Screen Shot 2020-01-15 at 17 56 12

### CODE REVIEW

File: lib/developer_portal/app/views/developer_portal/docs.html.liquid [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 <p>Use our live documentation to learn about the Echo API</p>
 
-{% cdn_asset /swagger-ui/2.2.10/swagger-ui.js %}
-{% cdn_asset /swagger-ui/2.2.10/swagger-ui.css %}
+{% cdn_asset /swagger-ui/3.24.3/swagger-ui-bundle.js %}
+{% cdn_asset /swagger-ui/3.24.3/swagger-ui.css %}
```
Comment (hallelujah) [inline lib/developer_portal/app/views/developer_portal/docs.html.liquid]: I think there will be incompatibilities with our custom JS code to render the user keys and such

File: lib/developer_portal/app/views/developer_portal/docs.html.liquid [hunk 2]
```diff
@@ -2,20 +2,8 @@
 
 <p>Use our live documentation to learn about the Echo API</p>
 
-{% cdn_asset /swagger-ui/2.2.10/swagger-ui.js %}
-{% cdn_asset /swagger-ui/2.2.10/swagger-ui.css %}
+{% content_for javascripts %}
+    {{ 'active_docs.js' | javascript_include_tag }}
```
Comment (damianpm) [inline lib/developer_portal/app/views/developer_portal/docs.html.liquid]: I'm not sure if importing this way is going to work, the content in `cdn_asset` is supposed to be retrieved from a cdn (our own).
Why are you moving it from the cdn?

Comment (josemigallas) [reply lib/developer_portal/app/views/developer_portal/docs.html.liquid]: The main reason is to keep track of the versioning. This way we no longer have to expose the entire bundle and migrate templates but wepacker/rails take care of it. It's supposed to work.. at least it does in local. @didierofrivia might be able to explain further (and better)

Comment (didierofrivia) [reply lib/developer_portal/app/views/developer_portal/docs.html.liquid]: Yeah, we are also aiming to get rid of modified 3rd party's assets (providing them with the cdn). Instead, having specific 3scale plugins (in the case of swagger specifically)

File: .circleci/config.yml [hunk 3]
```diff
@@ -440,8 +440,7 @@ jobs:
           name: Precompile assets
           command: |
             cp config/examples/*.yml config/
-            bundle exec rake assets:precompile RAILS_ENV=test
-            bundle exec rake assets:precompile RAILS_ENV=production
+            bundle exec rake assets:precompile RAILS_ENV=test WEBPACKER_PRECOMPILE=false
```
Comment (guicassolato) [inline .circleci/config.yml]: This doesn't seem to belong to this PR.

**Comment by guicassolato** [comment]
A spec that can be used to try it out:

```json
{
   "openapi": "3.0.0",
   "info": {
      "version": "1.0.0",
      "title": "Echo API",
      "description": "A sample echo API"
   },
   "paths": {
      "/": {
         "get": {
            "description": "Echo API with no parameters",
            "operationId": "echo_no_params",
            "parameters": [
               {
                  "name": "user_key",
                  "in": "query",
                  "description": "Your API access key",
                  "required": true,
                  "x-data-threescale-name": "user_keys",
                  "schema": {
                     "type": "string"
                  }
               }
            ],
            "responses": {
               "200": {
                  "description": "response",
                  "content": {
                     "application/json": {
                        "schema": {
                           "$ref": "#/components/schemas/ResponseModel"
                        }
                     },
                     "application/xml": {
                        "schema": {
                           "$ref": "#/components/schemas/ResponseModel"
                        }
                     },
                     "text/xml": {
                        "schema": {
                           "$ref": "#/components/schemas/ResponseModel"
                        }
                     },
                     "text/html": {
                        "schema": {
                           "$ref": "#/components/schemas/ResponseModel"
                        }
                     }
                  }
               },
               "default": {
                  "description": "unexpected error",
                  "content": {
                     "application/json": {
                        "schema": {
                           "$ref": "#/components/schemas/ErrorModel"
                        }
                     },
                     "application/xml": {
                        "schema": {
                           "$ref": "#/components/schemas/ErrorModel"
                        }
                     },
                     "text/xml": {
                        "schema": {
                           "$ref": "#/components/schemas/ErrorModel"
                        }
                     },
                     "text/html": {
                        "schema": {
                           "$ref": "#/components/schemas/ErrorModel"
                        }
                     }
                  }
               }
            }
         }
      },
      "/{echo}": {
         "get": {
            "description": "Echo API with parameters",
            "operationId": "echo_with_params",
            "parameters": [
               {
                  "name": "echo",
                  "in": "path",
                  "description": "The string to be echoed",
                  "required": true,
                  "schema": {
                     "type": "string"
                  }
               },
               {
                  "name": "user_key",
                  "in": "query",
                  "description": "Your API access key",
                  "required": true,
                  "x-data-threescale-name": "user_keys",
                  "schema": {
                     "type": "string"
                  }
               }
            ],
            "responses": {
               "200": {
                  "description": "response",
                  "content": {
                     "application/json": {
                        "schema": {
                           "$ref": "#/components/schemas/ResponseModel"
                        }
                     },
                     "application/xml": {
                        "schema": {
                           "$ref": "#/components/schemas/ResponseModel"
                        }
                     },
                     "text/xml": {
                        "schema": {
                           "$ref": "#/components/schemas/ResponseModel"
                        }
                     },
                     "text/html": {
                        "schema": {
                           "$ref": "#/components/schemas/ResponseModel"
                        }
                     }
                  }
               },
               "default": {
                  "description": "unexpected error",
                  "content": {
                     "application/json": {
                        "schema": {
                           "$ref": "#/components/schemas/ErrorModel"
                        }
                     },
                     "application/xml": {
                        "schema": {
                           "$ref": "#/components/schemas/ErrorModel"
                        }
                     },
                     "text/xml": {
                        "schema": {
                           "$ref": "#/components/schemas/ErrorModel"
                        }
                     },
                     "text/html": {
                        "schema": {
                           "$ref": "#/components/schemas/ErrorModel"
                        }
                     }
                  }
               }
            }
         }
      }
   },
   "servers": [
      {
         "url": "{protocol}://echo-api.3scale.net/",
         "variables": {
           "protocol": {
             "enum": ["http", "https"],
             "default": "http"
           }
         }
      }
   ],
   "components": {
      "schemas": {
         "ResponseModel": {
            "type": "object",
            "required": [
               "method",
               "path",
               "args",
               "headers"
            ],
            "properties": {
               "method": {
                  "type": "string"
               },
               "path": {
                  "type": "string"
               },
               "args": {
                  "type": "string"
               },
               "headers": {
                  "type": "object"
               }
            }
         },
         "ErrorModel": {
            "type": "object",
            "required": [
               "code",
               "message"
            ],
            "properties": {
               "code": {
                  "type": "integer",
                  "format": "int32"
               },
               "message": {
                  "type": "string"
               }
            }
         }
      }
   }
}
```

**Comment by josemigallas** [comment]
Closing in favor of #1594
